### PR TITLE
Account setup

### PR DESCRIPTION
**Pull request by mkyl**
This pull request is to fulfill technical deliverable 1, "Using the Google Drive API, the Android app will allow the user to authorize their Google account so the app can upload content."

Find a short video demo of this code here: https://youtu.be/HKnuS_gt0T4

### CODE REVIEW

File: app/src/main/java/com/physical_web/cms/physicalwebcms/DriveSetupActivity.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        busyAuthorizing = true;
+        findViewById(R.id.textViewWarning).setVisibility(View.INVISIBLE);
+        findViewById(R.id.indeterminateBar).setVisibility(View.VISIBLE);
+        findViewById(R.id.start_setup_button).setEnabled(false);
+    }
+
+    // enable interface interaction after a successful authorization
+    private void unlockInterface() {
+        findViewById(R.id.indeterminateBar).setVisibility(View.INVISIBLE);
+        this.findViewById(R.id.textViewWarning).setVisibility(View.INVISIBLE);
```
Comment (jelcaf) [inline app/src/main/java/com/physical_web/cms/physicalwebcms/DriveSetupActivity.java]: Why you are using `this.findById` and in some cases `findById` (without `this`) ?

File: app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    SharedPreferences sharedPreferences;
+
+    public SetupManager(Context context) {
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+    }
+
+    public void checkRequirements(Context context) {
+        if(isFirstRun()) {
+            Intent welcomeIntent = new Intent(context, WelcomeActivity.class);
+            context.startActivity(welcomeIntent);
```
Comment (jelcaf) [inline app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java]: If I run the code...  crash the app.

screen shot 2017-06-19 at 10 32 01 am

I'm not quite sure, but if you use `welcomeIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);` before start the new activity could fix the problem.

**Review by jelcaf — CHANGES_REQUESTED** [review]
Hi @mkyl, good job! But I can't run the code in my environment. In the comments you can find the request changes (one of them fix the crash in the app).

File: app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java (hunk 2 above)
Comment (mkyl) [reply app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java]: What version of Android are you running?

File: app/src/main/java/com/physical_web/cms/physicalwebcms/DriveSetupActivity.java (hunk 1 above)
Comment (mkyl) [reply app/src/main/java/com/physical_web/cms/physicalwebcms/DriveSetupActivity.java]: Fixed.

File: app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java (hunk 2 above)
Comment (mkyl) [reply app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java]: I think a545d15 should fix this bug

**Comment by mkyl** [comment]
Thanks for the feedback. Let me know if there is anything else to improve. 

Also I wanted to note that this branch doesn't really have tests because I thought any tests for drive sign-up would be difficult to setup and fragile/inconsistent. I've written tests for the other parts of the setup which are in a different branch.

**Comment by jelcaf** [comment]
Awesome!! You can merge it.

**Review by jelcaf — APPROVED** [review]

**Pull request merged**